content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -1,20 +1,468 @@
+        output.m_tangent = mul(PerObject::GetNormalMatrix(), input.m_tangent);
+       
+        return output;
+    }
+  
+### Structs
+Just like in HLSL, structs can be defined globally, but in AZSL they can also be defined and instantiated inside SRGs. You can only instantiate globally defined structs inside SRGs, rootconstants or inside another struct or class.  
+  
+In general, structs are supported just like in HLSL. struct has no constructors/destructors and as such, do not accept default member initializers.  
+  
+Just like in HLSL, struct has the particularity to be data only (no methods).
+
+### Classes
+Classes are supported in AZSL with the exact same expectations of how they are supported in HLSL. They can only be declared/defined in global scopes (outside of SRGs), but an SRG can instantiate class type variables.  
+  
+Like in HLSL, classes can inherit from interfaces to be forced to respect a minimal method set.  
+  
+Unlike in HLSL, AZSL forbids class inheritance among classes.  
+  
+Classes has no constructors/destructors and as such, do not accept default member initializers.
+
+### Matrices
+AZSL follows the HLSL convention to use column major matrices by default.  
+  
+However, in the O3DE Runtime, the default convention is row major matrices.  
+  
+When declaring a matrix, the order can be specified using the row_major or column_major keywords.  
+  
+    ShaderResourceGroupSemantic MatrixExample
+    {
+        FrequencyId = 1;
+    };
+      
+    ShaderResourceGroup MatrixExampleSRG : MatrixExample
+    {
+                     float2x3 defaultMatrix;
+        row_major    float2x3 rowMajorMatrix;
+        column_major float2x3 colMajorMatrix;
+    };
+  
+### Arrays And Unbounded Arrays
+From the point of view of defining arrays, the rules and limitations are exactly the same as HLSL. Examples:  
+  
+    ShaderResourceGroup PerObject : BindingPerObject
+    {
+        float m_arrayOfFloats[64]; // GOOD. Arrays of fundamental types must be of constant size.
+     
+        struct SomeStruct
+        {
+            int m_integer;
+            bool m_boolean;
+        }
+        static const int MyArraySize = 32;
+        SomeStruct m_arrayOfStructs[MyArraySize]; // GOOD. Arrays of packable types (structs & fundamentals) require constant size definition.
+     
+        int m_unboundedArrayOfIntegers[]; // ERROR. Unbounded arrays are not supported for fundamental(packables, in general) data types.
+     
+        Texture2D<float4> m_textures[MyArraySize]; // GOOD. Contant size array of Resource View types is ok.
+         
+        ConstantBuffer<SomeStruct> m_unboundedArrayOfStructs[]; // GOOD. Unbounded array of Resource View type is ok. BUT, there are some restrictions.
+    }
+  
+To learn more about what are the limitations when declaring unbounded arrays, go here: [Binding Rules For Unbounded Arrays.](unbounded-arrays)  
+  
+### <a name="partial"></a> Partial ShaderResourceGroup Definitions.
```

Comment:
- If you make this fix, please fix references to this section. For example:
   - Within the same page: [Partial Shader Resource Groups](#partial-shader-resource-groups)
   - Outside of this page: [Partial Shader Resource Groups](<path>/azsl/_index.md#partial-shader-resource-groups)
```suggestion
### Partial Shader Resource Groups
```

Reply:
Fixed.